src/chat/src/Bridge/MongoDb/MessageStore.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\AI\Chat\Bridge\MongoDb;
+
+use MongoDB\Client;
+use Symfony\AI\Chat\ManagedStoreInterface;
+use Symfony\AI\Chat\MessageNormalizer;
+use Symfony\AI\Chat\MessageStoreInterface;
+use Symfony\AI\Platform\Message\MessageBag;
+use Symfony\AI\Platform\Message\MessageInterface;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+use Symfony\Component\Serializer\Normalizer\ArrayDenormalizer;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Symfony\Component\Serializer\Serializer;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * @author Guillaume Loulier <personal@guillaumeloulier.fr>
+ */
+final class MessageStore implements ManagedStoreInterface, MessageStoreInterface
+{
+    public function __construct(
+        private readonly Client $client,
+        private readonly string $databaseName,
+        private readonly string $collectionName,
+        private readonly SerializerInterface&NormalizerInterface&DenormalizerInterface $serializer = new Serializer([
+            new ArrayDenormalizer(),
+            new MessageNormalizer(),
+        ], [new JsonEncoder()]),
+    ) {
+    }
+
+    public function setup(array $options = []): void
+    {
+        $database = $this->client->selectDatabase($this->databaseName);
+
+        $database->createCollection($this->collectionName, $options);
+    }
+
+    public function drop(): void
+    {
+        $this->client->getCollection($this->databaseName, $this->collectionName)->deleteMany([
+            'q' => [],
+        ]);
+    }
+
+    public function save(MessageBag $messages): void
+    {
+        $currentCollection = $this->client->getCollection($this->databaseName, $this->collectionName);
+
+        $currentCollection->insertMany(array_map(
+            fn (MessageInterface $message): array => $this->serializer->normalize($message),
+            $messages->getMessages(),
+        ));
+    }
+
+    public function load(): MessageBag
+    {
+        $currentCollection = $this->client->getCollection($this->databaseName, $this->collectionName);
+
+        return new MessageBag(...array_map(
+            fn (\ArrayObject $message): MessageInterface => $this->serializer->denormalize($message->getArrayCopy(), MessageInterface::class),
+            $currentCollection->find()->toArray(),
```

Comment:
I'm surprised there is no filter or sort order, but that seems to be how the interface is designed.

Reply:
In fact, we can sort them thanks to the `added_at` key (which store a timestamp) but by design, no sorting here, maybe something to improve @chr-hertel @OskarStark ?